cell_based/test/tutorial/TestRunningImmersedBoundarySimulationsTutorial.hpp

```diff
@@ -0,0 +1,159 @@
+ * Redistributions of source code must retain the above copyright notice,
+   this list of conditions and the following disclaimer.
+ * Redistributions in binary form must reproduce the above copyright notice,
+   this list of conditions and the following disclaimer in the documentation
+   and/or other materials provided with the distribution.
+ * Neither the name of the University of Oxford nor the names of its
+   contributors may be used to endorse or promote products derived from this
+   software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE
+GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
+OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+*/
+
+#ifndef TESTIMMERSEDBOUNDARYDEMOTUTORIAL_HPP_
+#define TESTIMMERSEDBOUNDARYDEMOTUTORIAL_HPP_
+
+/*
+ * = Example showing how to create and run an immersed boundary simulation in Chaste =
+ *
+ * We create a simple palisade of cells with a basement membrane, and see how to:
+ *   * set the initial conditions;
+ *   * change the cell-level properties.
+ *
+ * == The test ==
+ *
+ * We begin by including the necessary header files.
+ */
+
+/* Required for the test environment */
+#include <cxxtest/TestSuite.h>
+#include "AbstractCellBasedTestSuite.hpp"
+
+/* Required for core elements of the Chaste simulation environment */
+#include "CellsGenerator.hpp"
+#include "DifferentiatedCellProliferativeType.hpp"
+#include "OffLatticeSimulation.hpp"
+#include "SmartPointers.hpp"
+#include "UniformCellCycleModel.hpp"
+
+/* Required for the immersed boundary functionality */
+#include "ImmersedBoundaryLinearInteractionForce.hpp"
+#include "ImmersedBoundaryCellPopulation.hpp"
+#include "ImmersedBoundaryLinearMembraneForce.hpp"
+#include "ImmersedBoundaryMesh.hpp"
+#include "ImmersedBoundarySimulationModifier.hpp"
+#include "ImmersedBoundaryPalisadeMeshGenerator.hpp"
+
+/* Required for setting up the numerical method */
+#include "ForwardEulerNumericalMethod.hpp"
+#include <boost/make_shared.hpp>
+
+/*
+// This test is never run in parallel */
+#include "FakePetscSetup.hpp"
+
+/* In Chaste, every simulation is run as a 'test', and here we define a test class which inherits from
+ * {{{AbstractCellBasedTestSuite}}}. The class {{{AbstractCellBasedTestSuite}}} sets up various parameters for us.  Of
```

Comment:
All the comments in this file want to follow the new website markdown (I think 3 brackets is trac)